### DISCUSSION TITLE
Preferred way to get chain ID from connected chain not listed in config

### DISCUSSION THREAD

**Discussion by 0xdeployer** [wevm/wagmi#4189] [Q&A]
The docs state the following for `useChainId` 

> Only returns chain IDs for chains configured via createConfig's [chains](https://wagmi.sh/react/api/createConfig#chains) parameter.  If the active [connection](https://wagmi.sh/react/api/createConfig#connection) [chainId](https://wagmi.sh/react/api/createConfig#chainid-1) is not from a chain included in your Wagmi Config, useChainId will return the last configured chain ID. 

This is a bit strange to me since the user can be connected to a completely different chain than the one intended and their wallet will still try to execute the transaction. 

To get the actual chain the client is connected to I'm getting `connector` from `useAccount` and calling `getChainId`. Is there another method for getting the actual chain ID the client is connected to? 

```
export function useSwitchToHamChain() {
  const { isConnected, connector } = useAccount();
  const _chainId = useChainId();
  const [chainId, setChainId] = useState(_chainId);
  useEffect(() => {
    (async () => {
      const c = await connector?.getChainId();
      setChainId(c ?? _chainId);
      console.log(c);
    })();
  }, [isConnected, connector, _chainId]);
  const { switchChain, switchChainAsync } = useSwitchChain();
  const isHamChain = chainId === ham.id;

  const switchToHamChain = () => {
    const { id } = ham;
    switchChain({ chainId: id, ...ham });
  };

  const switchToHamChainAsync = async (fn?: () => void) => {
    if (!isHamChain) {
      const { id } = ham;
      await switchChainAsync({ chainId: id, ...ham });
    }

    fn?.();
  };

  return {
    isHamChain,
    switchToHamChain,
    switchToHamChainAsync,
  };
}
```

**Comment by tmm** [comment]
`useAccount().chainId` should be the same as `useAccount().connector?.getChainId()`

**Comment by flymauntains** [comment]
Need wagmi version, According to wagmi version, method get chainId is different.
